### ISSUE TITLE
make sure cc.map static maps display in TkAgg, QT, and various other backends

### ISSUE THREAD

**Issue by andy-esch** [CartoDB/cartoframes#248]
The following should display the output in the proper framework.
```
ipython --pylab
>>> cc.map(..., interactive=False)
```

**Comment by michellemho** [comment]
I haven't been test this thoroughly, but these are a few investigations I did, and they seemed to work:

```
# Expected behavior: saves map as a PDF
import matplotlib
matplotlib.use('PDF')
import cartoframes
cc = cartoframes.CartoContext()
cc.map(interactive=False)
import matplotlib.pyplot as plt
plt.savefig('blank_map')

# Expected behavior: saves map as a SVG
import matplotlib
matplotlib.use('SVG')
import cartoframes
cc = cartoframes.CartoContext()
cc.map(interactive=False)
import matplotlib.pyplot as plt
plt.savefig('blank_map')

# Expected behavior: saves map as a PNG
import matplotlib
matplotlib.use('agg')
import cartoframes
cc = cartoframes.CartoContext()
cc.map(interactive=False)
import matplotlib.pyplot as plt
plt.savefig('blank_map')
```

I wasn't able to test other backends (interactive backends that render to a canvas, not just a file) because they require other packages to be downloaded and installed. I wasn't able to figure those out. These included `CAIRO`, `WX`, `GTK`, `QT4Agg`.

**Comment by andy-esch** [comment]
I got it to work with the `osx` backend too. Since I use virtual environments I've had to do some extra setup ([IPython tips](http://ipython.readthedocs.io/en/stable/interactive/plotting.html), and [matplotlib tips](https://matplotlib.org/faq/osx_framework.html)).

After I got that bash function running I get my IPython going with `frameworkpython -m IPython` (where I altered the bash function to be python3 only).

Then running this:
```
In [1]: import cartoframes
In [2]: from cartoframes import Layer
In [3]: cc = cartoframes.CartoContext(...)
In [4]: cc.map(layers=Layer('all_month_3', color='net')
```
Which throws up a figure in a separate window using the `osx` backend:
<img width="1434" alt="screen shot 2017-10-27 at 11 14 24" src="https://user-images.githubusercontent.com/1041056/32112264-c4fe8094-bb0a-11e7-862e-436f1258c6b2.png">
